sdk/go/common/workspace/plugins.go

```diff
@@ -1495,26 +1496,28 @@ func cleanupTempDirs(finalDir string) error {
 	return nil
 }
 
-// PluginKind represents a kind of a plugin that may be dynamically loaded and used by Pulumi.
-type PluginKind string
+// Re exporting PluginKind to keep backward compatibility, this should be kept in sync with
+// the definitions in sdk/go/common/apitype/plugins.go
+//
+// Deprecated: PluginKind type was moved to "github.com/pulumi/pulumi/sdk/v3/go/common/apitype"
+type PluginKind = apitype.PluginKind
 
+// ProductKind types definition
+//
+// Deprecated: PluginKind type was moved to "github.com/pulumi/pulumi/sdk/v3/go/common/apitype"
 const (
-	// AnalyzerPlugin is a plugin that can be used as a resource analyzer.
-	AnalyzerPlugin PluginKind = "analyzer"
-	// LanguagePlugin is a plugin that can be used as a language host.
-	LanguagePlugin PluginKind = "language"
-	// ResourcePlugin is a plugin that can be used as a resource provider for custom CRUD operations.
-	ResourcePlugin PluginKind = "resource"
-	// ConverterPlugin is a plugin that can be used to convert from other ecosystems to Pulumi.
-	ConverterPlugin PluginKind = "converter"
-	// ToolPlugin is an arbitrary plugin that can be run as a tool.
-	ToolPlugin PluginKind = "tool"
+	AnalyzerPlugin  = apitype.AnalyzerPlugin
+	LanguagePlugin  = apitype.LanguagePlugin
+	ResourcePlugin  = apitype.ResourcePlugin
+	ConverterPlugin = apitype.ConverterPlugin
+	ToolPlugin      = apitype.ToolPlugin
 )
 
 // IsPluginKind returns true if k is a valid plugin kind, and false otherwise.
 func IsPluginKind(k string) bool {
```

Comment:
Feels maybe a little bit odd not to have this in `apitype` as well, but I don't think it's worth moving unless we actually have to.